src/sentry/migrations/0378_backfill_perf_issue_events_issue_platform.py

```diff
@@ -0,0 +1,555 @@
+    attempted_row: TransactionRow,
+    exc: Exception,
+    occurrence_nodestore_saved: bool,
+    occurrence_eventstream_sent: bool,
+) -> None:
+    print(  # noqa: S002
+        f"row={attempted_row}, exception={exc}, nodestore_saved={occurrence_nodestore_saved}, eventstream_sent={occurrence_eventstream_sent}"
+    )
+
+
+def backfill_eventstream(apps: Any, schema_editor: Any) -> None:
+    """
+    Backfills Performance-issue events from the transaction table to the IssuePlatform dataset(search_issues).
+    1. Source Groups from postgres as the 'source of truth' for issues.
+    2. For each project with a performance issue:
+         a. Query snuba for the transactions backing a perf issue. We only need (project_id, group_id, event_id).
+            This is paginated with a limit of 10000 transactions per page to avoid pull in too much data into memory.
+         b. Query snuba for generic events backing a perf issue (we'll be dual-writing to both tables, so we need
+            to ensure the event doesn't already exist in search_issues).
+         c. For each transaction:
+              i. Retrieve the Group row from postgres.
+              ii. Retrieve the GroupHash row from postgres.
+              iii. Retrieve the entire raw Transaction from node store.
+              iv.
+              v. Map the Transaction event to an instance of IssueOccurrenceData.
+              v. Save the IssueOccurrence to node store. Normally this step would also save the appropriate models to
+                 postgres like Group, GroupHash, GroupEnvironment, GroupRelease, and increment Release counts, but
+                 we aren't doing that on this backfill since they should already exist when the transaction was already
+                 ingested through the old style of performance issue creation.
+              vi. Send the IssueOccurrence through the eventstream to be saved in search_issues.
+         c. Repeat step (a), until the paginated snuba query returns no data.
+    """
+
+    Group = apps.get_model("sentry", "Group")
+    GroupHash = apps.get_model("sentry", "GroupHash")
+
+    projects_with_perf_issues = (
+        Group.objects.filter(type__in=PERFORMANCE_TYPES)
+        .values("project_id")
+        .order_by("project_id")
+        .distinct()
+    )
+
+    for project_perf_issue in projects_with_perf_issues:
+        backfill_by_project(project_perf_issue["project_id"], Group, GroupHash, DRY_RUN)
+
+
+class SuccessProgress(typing.TypedDict):
+    occurrence_id: str
+    event_id: str
+    fingerprint: str
+
+
+class ErrorProgress(typing.TypedDict):
+    event_id: str
+    exception: str
+    nodestore_saved: bool
+    eventstream_sent: bool
+
+
+class ProjectGroupTotalProgress:
+    """
+    1 success row ~= 150 bytes
+    1 error row ~= 150 bytes
+
+    The performance issue with the most events is around 9M transactions. So the largest file for a group should
+    be capped at around 1.28 GB ...
+    """
```

Comment:
Was doing some napkin paper math and this approach of writing the processing status of each transaction to a per-project-group file may blow up memory and/or get really hairy with open file handles. Maybe I should use redis instead?